OpenSim/Actuators/TorqueActuator.cpp

```diff
@@ -191,13 +191,16 @@ void TorqueActuator::extendConnectToModel(Model& model)
             "TorqueActuator::extendConnectToModel(): body name properties "
             "were not set.");
 
-    // Look up the bodies by name in the Model, and record pointers to the
-    // corresponding body objects.
-    _bodyA =
-        static_cast<const PhysicalFrame*>(&getModel().getComponent(get_bodyA()));
-    _bodyB =
-        static_cast<const PhysicalFrame*>(&getModel().getComponent(get_bodyB()));
-
+    // TODO: Replace this custom lookup with Sockets
+    if(getModel().hasComponent<PhysicalFrame>(get_bodyA()))
+        _bodyA = &getModel().getComponent<PhysicalFrame>(get_bodyA());
+    else
```

Comment:
The sequence of hasComponent<> then getComponent<> now appears in many places, it likely does the work twice, can these be combined for mentainance and efficiency?

Reply:
Also I'm not sure if the first hasComponent<> would succeed even if the Body is under  bodyset, or does it search only immediate children?

Reply:
> The sequence of hasComponent<> then getComponent<> now appears in many places, it likely does the work twice, can these be combined for mentainance and efficiency?

The code is not duplicated. They both share the same worker method, `traversePathToComponent()`. I am inclined not to expose  `traversePathToComponent()` unless we can show that the current interface has a performance issue.